BlogTemplate/Pages/Account/ConfirmEmail.cshtml.cs

```diff
@@ -28,13 +28,13 @@ public ConfirmEmailModel(UserManager<ApplicationUser> userManager)
             var user = await _userManager.FindByIdAsync(userId);
             if (user == null)
             {
-                throw new ApplicationException($"Unable to load user with ID '{userId}'.");
+                throw new FieldAccessException($"Unable to load user with ID '{userId}'.");

```

Comment:
It's still possible for a user to be not found - for example, I log in, my user id is `abc123` (or whatever), but then I navigate to the URL `/Account/ConfirmEmail?userId=xyz789&code=zzzzzzz` Assuming user id `xyz789` doesn't exist, some error will need to occur.

Speaking of `BadRequestException`, I can't find any exception of that name outside of IdentityModel (which wouldn't be appropriate here - it's for IdentityModel's own exceptions) - which were you thinking?